src/settings.js

```diff
@@ -1217,6 +1217,14 @@ var WASM = 1;
 // environments the expectation is to create the memory in the wasm itself.
 // Doing so prevents some possible JS optimizations, so we only do it behind
 // this flag.
+//
+// When this flag is set we do not legalize the JS interface, since the wasm is
+// meant to run in a wasm VM, which can handle i64s directly. If we legalized it
+// the wasm VM would not recognize the API. However, this means that the
+// optional JS emitted won't run if you use a JS API with an i64. You can use
+// the WASM_BIGINT option to avoid that problem by using BigInts for i64s which
+// means we don't need to legalize for JS (but this requires a new enough JS
+// VM).
```

Comment:
Shall we just say that STANDALONE_WASM implies WASM_BIGINT ?    Is STANDALONE_WASM not completely broken without it?

Reply:
I considered this, yeah, it's tempting to do that. The problem is that it is *not* completely broken without it... only when there is an actual i64 on the ffi boundary. That's why it matters for this clock test, but no other ones. So it feels a little awkward to force bigint support in standalone mode all the time.

Reply:
I tempted to say lets not try to make every possible configuration work.   Standalone requiring bigint seems fine to me at least until we have users wanting this config.     Surely there are other syscalls that needs bigint too, no?   I think supporting this extra half-way-house mode is not worth the effort.

If we do wan to support this method we should ifdef out the i64 syscalls sompletely so such users get link errors an not runtime errors when try try to link in an i64-using syscall . (is that true today?)

How about at least enabling WASM_BIGINT by default in standalone mode but allow people to explicitly opt out with WASM_BIGINT=0 (and give them link errors when they use the wrong syscalls).

Reply:
I agree fewer combinations is better. I'm not sure however which way is best to achieve that.

In particular I think it adds complexity to make an option automatically turn on another option (or change its default). It means more things to think about when a bug report comes in. There are cases where it's worth it, but I don't think this is one of them.

Ifdefing out i64-using syscalls is an interesting option that you mention. Or we could make it a link error to have them included. I lean towards that myself.

Reply:
I still think its fine to assume that we have bigint support by default in standalone wasm mode myself.    I think this will be especially true as more 64-bit-using syscalls get added.

If we don't do this aren't we going to have all of the users of STANDALONE_WASM mode who want to run in JS be forced to add this second option?  

What does the error message look like if I forget to build my standalone program with WASM_BIGINT today?  Will it be obvious from the error message how to fix it?  

(apologies for basically repeating myself, I'm fine landing this change either way, we can always evolve this over time).

Reply:
Right now if you build without bigint support but with standalone (so no legalization), it will work ok if no ffi is called with an i64. If one is called, the wasm vm throws, `illegal type in function signature` or such. That's not terribly clear, but if you are aware of the general issue it makes sense.

Or, if you build with bigint support and run in a wasm VM without it, you will get the exact same error.

Yeah, I'm not totally happy with that either way, really. But probably most standalone users are not running the JS (standalone helps avoid the JS).